### PR TITLE
Remove MMX from Rust 

### PR DESCRIPTION
**Pull request by mati865**
Follow-up to https://github.com/rust-lang/stdarch/pull/890
This removes most of MMX from Rust (tests pass with small changes), keeping stable `is_x86_feature_detected!("mmx")` working.

### CODE REVIEW

**Comment by mati865** [comment]
cc @Amanieu

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs [hunk 1]
```diff
@@ -233,10 +233,9 @@ pub enum TypeKind {
     Pointer = 12,
     Vector = 13,
     Metadata = 14,
-    X86_MMX = 15,
```
Comment (nikic) [inline compiler/rustc_codegen_llvm/src/llvm/ffi.rs:236]: Don't think we should remove it from the LLVM FFI API...

Comment (mati865) [reply compiler/rustc_codegen_llvm/src/llvm/ffi.rs:236]: I agree but on the other hand it's basically dead code.

File: compiler/rustc_codegen_llvm/src/type_of.rs [hunk 2]
```diff
@@ -20,23 +20,16 @@ fn uncached_llvm_type<'a, 'tcx>(
     match layout.abi {
         Abi::Scalar(_) => bug!("handled elsewhere"),
         Abi::Vector { ref element, count } => {
+            // FIXME: Replace this comment
```
Comment (Amanieu) [inline compiler/rustc_codegen_llvm/src/type_of.rs:23]: I think we can just remove this comment. The previous MMX hack would likely have caused issues with `packed_simd` anyways.

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs (hunk 1 above)
Comment (Amanieu) [reply compiler/rustc_codegen_llvm/src/llvm/ffi.rs:236]: This enum is meant to match `LLVMTypeKind` in the LLVM C API.

File: compiler/rustc_codegen_llvm/src/type_of.rs (hunk 2 above)
Comment (mati865) [reply compiler/rustc_codegen_llvm/src/type_of.rs:23]: Like [that one](https://github.com/rust-lang/rust/issues/53254) affecting windows-gnu and apple?

**Comment by mati865** [comment]
Updated, I'll squash commits afterwards.

**Comment by danieldjewell** [comment]
Can someone please double check the above referenced issues in rust-lang/packed_simd and ripgrep? If these changes did make it into nightly on 2020-09-15 then this change would appear to be the root cause of the compilation failures.

**Comment by Amanieu** [comment]
LGTM but probably needs a sign-off from @rust-lang/compiler.

**Comment by mati865** [comment]
Don't you want me to squash commits first?

I left them to make review easier but this causes mess in git history.

**Comment by oli-obk** [comment]
@bors r-

@bors delegate+

Sure, feel free to `r=Amanieu,oli-obk` after squashing

**Pull request merged**